src/itemsList/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { Component, FunctionComponent } from 'react'
+import cc from 'classcat'
+
+import { ItemChoiceProps } from 'itemChoice'
+import { ItemRadioProps } from 'itemRadio'
+
+import style from './style'
+
+export const ItemsListDivider: FunctionComponent = () => null
+
+type ItemsListChild =
+  | React.ReactElement<ItemChoiceProps>
+  | React.ReactElement<ItemRadioProps>
+  | null
+
+export interface ItemsListProps {
+  readonly children: ItemsListChild[]
+  readonly withSeparators?: boolean
+  readonly className?: Classcat.Class
+  readonly keyGenerator?: (index: number) => string | number
+  readonly role?: string
+}
+
+class ItemsList extends Component<ItemsListProps> {
+  static defaultProps: Partial<ItemsListProps> = {
+    withSeparators: false,
+    className: '',
+    role: '',
+    keyGenerator: index => index,
+  }
+
+  static Divider = ItemsListDivider
+
+  render() {
+    const { children, className, withSeparators, keyGenerator, ...otherProps } = this.props
+    const { list } = React.Children.toArray(children).reduce(
```

Comment:
The following block is kinda hard to read. Could you split it a bit?

Reply:
I changed the approach (less `.reduce()` which I know isn't easy to read for a lot of people).